dev/PushNotifications/PushNotificationsLongRunningTask/ForegroundSinkManager.cpp

```diff
@@ -27,7 +24,7 @@ bool ForegroundSinkManager::InvokeForegroundHandlers(std::wstring const& process
         BOOL foregroundHandled = true;
         if (FAILED(it->second->InvokeAll(payloadSize, payload.data(), &foregroundHandled)))
```

Comment:
There is a bug in the foregroundsink map. All the foregroundsinks need to be stored as AgileRef's in the map so that we can invoke the main process from a different apartment context. Otherwise if the ForegroundSink was intialized in a different apartment from the callback invocation thread, all calls will end up failing.

Reply:
Yeah, I noticed that comment in the WNP_LRP PR. @pmpurifoy will address it.